packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts

```diff
@@ -0,0 +1,167 @@
+import {
+  extensionArraySchema,
+  remoteEntryScript,
+  remoteEntryCallback,
+} from '@openshift/dynamic-plugin-sdk';
+import type { Extension } from '@openshift/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import * as fs from 'fs';
+import * as glob from 'glob';
+import * as path from 'path';
+import * as webpack from 'webpack';
+import type { PluginBuildMetadata } from '../types/plugin';
+import { pluginBuildMetadataSchema } from '../yup-schemas';
+import { GenerateManifestPlugin } from './GenerateManifestPlugin';
+import { PatchContainerEntryPlugin } from './PatchContainerEntryPlugin';
+
+const parseJSONFile = (filePath: string) => JSON.parse(fs.readFileSync(filePath, 'utf-8'));
+
+const loadPluginMetadata = (fileName: string, baseDir = process.cwd()) => {
+  const filePath = path.resolve(baseDir, fileName);
+  return validatePluginMetadata(parseJSONFile(filePath));
+};
+
+const loadExtensions = (globPattern: string, baseDir = process.cwd()) => {
+  const filePaths = glob.sync(globPattern, { cwd: baseDir, absolute: true, nodir: true });
+  return _.flatMap(filePaths.map((filePath) => validateExtensions(parseJSONFile(filePath))));
+};
+
+const validatePluginMetadata = (obj: unknown): PluginBuildMetadata =>
+  pluginBuildMetadataSchema.strict(true).validateSync(obj);
+
+const validateExtensions = (obj: unknown): Extension[] =>
+  extensionArraySchema.strict(true).validateSync(obj);
```

Comment:
I do not believe I'd expect a validate function to return the data. I'm asking "is this valid" not "and give me back the data if it is". 🤔 

I get you're also using this as a typeguard, but it seems odd to me.

Reply:
I've improved the code to separate the parse vs. validate concerns better.